src/main/scala/sigmastate/serialization/SigmaSerializer.scala

```diff
@@ -72,6 +75,8 @@ trait SigmaSerializerCompanion[TFamily] {
   def getSerializer(opCode: Tag): SigmaSerializer[TFamily, _ <: TFamily]
 
   final def deserialize(bytes: Array[Byte], pos: Position): (TFamily, Consumed) = {
+    if (bytes.length > Serializer.MaxInputSize)
+      throw new InputSizeLimitExceeded(s"input size ${bytes.length} exceeds ${ Serializer.MaxInputSize}")
```

Comment:
@greenhat we still have this old-style methods? Let's get rid of them completely. Even if we need the methods like this, they should be wrappers and they shouldn't have any logic other than wrapping.